lsp/ts_ls.lua

```diff
@@ -44,7 +52,7 @@ return {
     'typescriptreact',
     'typescript.tsx',
   },
-  root_markers = { 'tsconfig.json', 'jsconfig.json', 'package.json', '.git' },
+  root_markers = { 'package-lock.json', 'yarn.lock', 'pnpm-lock.yaml', 'bun.lockb' },
```

Comment:
Hmm. Interesting. Thanks for carefully thinking about this. To avoid future regressions, it's worth a short comment giving a hint about why tsconfig etc. are not here.
```suggestion
  -- tsconfig.json, package.json are intentionally omitted because
  -- they can be "false positives" in a monorepo.
  root_markers = { 'package-lock.json', 'yarn.lock', 'pnpm-lock.yaml', 'bun.lockb' },
```